salt/state.py

```diff
@@ -2507,22 +2508,16 @@ def _get_envs(self):
             envs.extend([x for x in list(self.opts['file_roots'])
                          if x not in envs])
         env_order = self.opts.get('env_order', [])
+        # Remove duplicates
+        env_order = list(collections.OrderedDict.fromkeys(env_order))
```

Comment:
May I ask why we're using this approach? There are many less-complicated ways of removing duplicates. Not to mention the fact that this method seems to be incredibly inefficient:

```python
>>> import timeit
>>> prep = 'env_order = "[\'foo\', \'bar\', \'foo\', \'baz\']"; import collections'
>>> t1 = timeit.Timer('list(set(env_order))', prep)
>>> t2 = timeit.Timer('list(collections.OrderedDict.fromkeys(env_order))', prep)
>>> t1.repeat()
[1.3065440654754639, 1.2755541801452637, 1.2373528480529785]
>>> t2.repeat()
[22.5503408908844, 25.115081071853638, 24.109618186950684]
```